master/buildbot/db/changes.py

```diff
@@ -221,6 +221,8 @@ def thd(conn):
             changes_tbl = self.db.model.changes
             q = changes_tbl.select(
                 whereclause=(changes_tbl.c.sourcestampid == sourcestampid))
+            # if there are multiple changes for this ssid, get the most recent one
+            q = q.order_by(changes_tbl.c.changeid.desc())
```

Comment:
I think if we are going to fetch only one, then we should also do the ```.limit(1)```

Reply:
Done. Keep in mind that this is will only occur when rebuilding the same revision.